### PR TITLE
Implement region filtering

### PR DESCRIPTION
**Pull request by Xpl0itU**

### CODE REVIEW

File: src/config.c [hunk 1]
```diff
@@ -53,6 +53,16 @@
 #define LANG_TCH	"Traditional chinese"
 #define LANG_SYS	"System settings"
 
+#define SET_EUR		"Europe"
+#define SET_USA		"USA"
+#define SET_JPN		"Japan"
+#define SET_ALL		"All"
+
+const int regEUR = 1;
```
Comment (V10lator) [inline src/config.c:61]: Why a list of cont ints instead of an enum?

File: src/config.c [hunk 2]
```diff
@@ -395,6 +430,38 @@ Swkbd_LanguageType getUnfilteredLanguage()
 	return lang;
 }
 
+int getRegion()
+{
+	return regionSetting;
+}
+
+char* getFormattedRegion(int region) {
+	if(region == regAll) {
```
Comment (V10lator) [inline src/config.c:439]: Why not switch/case?

File: src/menu/configMenu.c [hunk 3]
```diff
@@ -88,24 +88,34 @@ void configMenu()
...
-/*		if(vpad.trigger & VPAD_BUTTON_LEFT)
+		if(vpad.trigger & VPAD_BUTTON_LEFT)
 		{
-			int l = getUnfilteredLanguage();
-			if(--l < Swkbd_LanguageType__Japanese)
-				l = Swkbd_LanguageType__Invalid;
-			
-			setKeyboardLanguage(l);
+			char* l = getFormattedRegion(getRegion());
+			if(l == "All") {
```
Comment (V10lator) [inline src/menu/configMenu.c:94]: What you do here (`==` with strings) is undefined behaviour. You could compare the `#define`d strings with `==` instead through, like this: `if(l == SET_ALL)`.

File: src/menu/mainMenu.c [hunk 4]
```diff
@@ -58,7 +58,7 @@ void drawMainMenuFrame()
 	textToFrame(7, MAX_CHARS - 27, "Thanks to:");
 	textToFrame(8, MAX_CHARS - 26, "• cJSON");
 	textToFrame(9, MAX_CHARS - 26, "• E1ite007");
-	textToFrame(10, MAX_CHARS - 26, "• libgui-sdl");
```
Comment (V10lator) [inline src/menu/mainMenu.c:61]: This should be a seperate pull request.

File: src/titles.c [hunk 5]
```diff
@@ -355,18 +357,22 @@ bool initTitles()
...
-			
-			tid = retransformTidHigh(i);
-			tid <<= 32;
-			tid |= 0x0000000010000000;
-			tid |= j;
-//			debugPrintf("%s / 0x%016X / 0x%016X", sj, j, tid);
-			titleEntry[entries].tid = tid;
-			++entries;
-			++titleEntries[cat];
+			if(titleEntry[entries].region == currentRegion) {
```
Comment (V10lator) [inline src/titles.c:360]: Are you sure you don't want to do `if(titleEntry[entries].region | currentRegion)` here?

File: src/titles.c [hunk 6]
```diff
@@ -355,18 +357,22 @@ bool initTitles()
...
+				
+				tid = retransformTidHigh(i);
+				tid <<= 32;
+				tid |= 0x0000000010000000;
+				tid |= j;
+	//			debugPrintf("%s / 0x%016X / 0x%016X", sj, j, tid);
+				titleEntry[entries].tid = tid;
+				++entries;
+				++titleEntries[cat];
+			} else {
```
Comment (V10lator) [inline src/titles.c:373]: No need for this else nor the continue as we're at the end of the loop anyway.

**Review by V10lator — CHANGES_REQUESTED** [review]
* All `char* ` to `char *`, please.
* Don't change version.
* Newline between `if` and `{`, please.

File: src/menu/configMenu.c (hunk 3 above)
Comment (Xpl0itU) [reply src/menu/configMenu.c:94]: I think I did this for better code readability and for displaying the region on the menu (?), I’m not totally sure tho

**Pull request merged**